### PR TITLE
Create source tarballs for releases.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/ethereum/solidity/issues/1183

### CODE REVIEW

File: scripts/create_source_tarball.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+set -e
+
+REPO_ROOT="$(dirname "$0")"/..
+(
+    cd "$REPO_ROOT"
+    version=`grep -oP "PROJECT_VERSION \"?\K[0-9.]+(?=\")"? CMakeLists.txt`
+    commithash=`git rev-parse --short=8 HEAD`
+    committimestamp=`git show --format=%ci HEAD | head -n 1`
+    commitdate=`git show --format=%ci HEAD | head -n 1 | cut - -b1-10 | sed -e 's/-0?/./' | sed -e 's/-0?/./'`
```
Comment (chfast) [inline scripts/create_source_tarball.sh]: Can't you use $committimestamp here?

File: scripts/create_source_tarball.sh [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env sh
+#
+
+set -e
+
+REPO_ROOT="$(dirname "$0")"/..
+(
+    cd "$REPO_ROOT"
+    version=`grep -oP "PROJECT_VERSION \"?\K[0-9.]+(?=\")"? CMakeLists.txt`
```
Comment (chfast) [inline scripts/create_source_tarball.sh]: I've read somewhere that `$()` is more portable than ` `` `.

File: scripts/create_source_tarball.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    commithash=`git rev-parse --short=8 HEAD`
+    committimestamp=`git show --format=%ci HEAD | head -n 1`
+    commitdate=`git show --format=%ci HEAD | head -n 1 | cut - -b1-10 | sed -e 's/-0?/./' | sed -e 's/-0?/./'`
+
+    # file exists and has zero size -> not a prerelease
+    if [ -e prerelease.txt -a ! -s prerelease.txt ]
```
Comment (chfast) [inline scripts/create_source_tarball.sh:14]: This is a bit confusing. How is this file managed?

File: .travis.yml [hunk 4]
```diff
@@ -142,15 +144,15 @@ cache:
...
-    - test "$TRAVIS_PULL_REQUESTS" != "false" || test "$TRAVIS_BRANCH" != release || echo -n > prerelease.txt # this is a proper release
 before_script:
     - test $TRAVIS_EMSCRIPTEN != On || ./scripts/build_emscripten.sh
     - test $TRAVIS_RELEASE != On || (mkdir -p build
       && cd build
       && cmake .. -DCMAKE_BUILD_TYPE=$TRAVIS_BUILD_TYPE
       && make -j2
       && cd ..
-      && ./scripts/release.sh $ZIP_SUFFIX )
+      && ./scripts/release.sh $ZIP_SUFFIX
```
Comment (chfast) [inline .travis.yml:154]: Why is this done before building and testing? Can be moved to "deploy"?

Comment (chriseth) [reply .travis.yml:154]: this only creates the artefacts, so I would say it is fine. The deployment itself is done in the `deploy` section.

File: scripts/create_source_tarball.sh (hunk 2 above)
Comment (chriseth) [reply scripts/create_source_tarball.sh]: ok, will change.

File: scripts/create_source_tarball.sh (hunk 1 above)
Comment (chriseth) [reply scripts/create_source_tarball.sh]: acutally `committimestamp` is not used at all.

File: scripts/create_source_tarball.sh (hunk 3 above)
Comment (chriseth) [reply scripts/create_source_tarball.sh:14]: The release branch has that file and it is empty. If the file does not exist or if it contains a string, it is a prerelease (in the former case, the "prerelease portion" of the version is composed from the commit hash, in the latter, the contents of that file are used directly).

**Review by chfast — APPROVED** [review]

**Pull request merged**